### ISSUE TITLE
worker.stop does not stop the worker (with pyzeebe 3.0.0rc3)

### ISSUE THREAD

**Issue by tomas-zemres** [camunda-community-hub/pyzeebe#197]
**Describe the bug**
We have simple example that runs worker and it should stop it after 5 seconds, but it is never stopped.

**The tested code**
```
import asyncio
import signal
from pyzeebe import ZeebeWorker, create_insecure_channel

async def main():
    channel = create_insecure_channel("zeebe", port=26500)
    worker = ZeebeWorker(channel)

    @worker.task(task_type="my_task")
    async def my_task(x: int): 
       return {"y": x + 1}

    async def stop_worker(*args, **kwargs):
        await asyncio.sleep(5)
        print("worker.stop")
        await worker.stop()
        print("end of stop_worker")

    async def run_worker():
        print("worker.work")
        await worker.work()
        print("end of run_worker")

    print("start")
    work_task = asyncio.create_task(run_worker())
    stop_task = asyncio.create_task(stop_worker())
    await work_task
    await stop_task
    print("end of main")


asyncio.run(main())

```

**It prints following output**
```
start
worker.work
worker.stop
end of stop_worker
```

Script is never ended and message "end of run_worker" is never printed.

**Versions:**
 - Python 3.8 in docker/linux
 - pyzeebe 3.0.0rc3
 
**Guess the problem**
Isn't the code stucked on the:
 ```job = await self.get_next_job()```
 until the next job is added to queue
 see:
 https://github.com/camunda-community-hub/pyzeebe/blob/pre-release/3.0.0/pyzeebe/worker/job_executor.py#L24

( it looks like the self.stop_event.is_set() is verified only after the job is executed, but if the job is not received, the test self.should_execute() is not called and the executor is not stopped )

**Comment by JonatanMartens** [comment]
Hey @tomas-zemres. First of all, thank you for bringing this to my attention!

I have opened a pull request that should fix this issue (#199). I plan to include this in version 3.0.0rc4. If you could in the meantime confirm that this fixes your problem, that would be great!

**Comment by tomas-zemres** [comment]
> Hey @tomas-zemres. First of all, thank you for bringing this to my attention!
> 
> I have opened a pull request that should fix this issue (#199). I plan to include this in version 3.0.0rc4. If you could in the meantime confirm that this fixes your problem, that would be great!

Yes, this code fixes the problem. Thank you.

**Comment by JonatanMartens** [comment]
Fixed in v3.0.0rc4
